Documentation of new spectral cube functionality in specutils (#784)
astropy/specutils: docs/spectral_cube.rst

Working with Spectral Cubes
###########################

Spectral cubes can be read directly with :class:`~specutils.Spectrum1D`.
A specific example of this is demonstrated here.

Note that the workflow described here is for spectral cubes that are rectified
such that one of the axes is entirely spectral and all the spaxels have the same

> ``spectral_axis`` values - i.e., case 2 in :ref:`specutils-representation-overview`.

Reviewer: ```suggestion
``spectral_axis`` values (i.e., case 2 in :ref:`specutils-representation-overview`).
```
I happened to notice the other day that the Astropy docs style guide says to put "i.e., ..." in parentheses.

For less-rectified cubes, pre-processing steps (not addressed by specutils at the
time of this writing) will be necessary to rectify the cubes into that form.

Loading a cube
==============

> We'll use the example cube data ``MaNGA cube``, and load the data from the

Reviewer: ```suggestion
We'll use a ``MaNGA cube`` for our example, and load the data from the
```

                      [4090.8855, 3179.1702, 2851.823 ],
                      [4158.919 , 3457.0115, 2841.1965]],

                     [[3684.6013, 3056.2   , 2880.6592],
                      [3221.7888, 2801.3518, 2525.541 ],
                      [3936.68  , 3461.534 , 3047.6135]]]))>

Spectral Bounding Region
========================

The `~specutils.manipulation.extract_bounding_spectral_region` function can be used to
extract the bounding region that encompases a set of disjoint `~specutils.spectra.SpectralRegion`
instances, or a composite instance of `~specutils.spectra.SpectralRegion` that contains
disjoint sub-regions.

.. code-block:: python

    >>> from specutils import SpectralRegion
    >>> from specutils.manipulation import extract_bounding_spectral_region
    >>> composite_region = SpectralRegion([(5000*u.AA, 5002*u.AA), (5006*u.AA, 5008.*u.AA)])
    >>> sub_spectrum = extract_bounding_spectral_region(sc, composite_region)
    >>> sub_spectrum.spectral_axis  #doctest:+SKIP
    [5000.3453, 5001.4969, 5002.6486, 5003.8007, 5004.953, 5006.1055, 5007.2584]A˚

Moments
=======

The `~specutils.analysis.moment` function can be used to compute moments of any order

> along one of the cube's axis. By default, ``axis=2`` (or alternatively, ``axis=-1``)
> is used to compute moments along the spectral axis.

Reviewer: ```suggestion
along one of the cube's axes. By default, ``axis=-1``, which computes moments 
along the spectral axis (remember that the spectral axis is always last in a 
:class:`~specutils.Spectrum1D`).
```
I think we can drop the ``axis=2``, since it's actually `-1` in the code. I thought the reminder that the spectral axis is always last would be useful to include here, but if it seems extraneous feel free to remove it.

Working with Spectral Cubes
###########################

Spectral cubes can be read directly with :class:`~specutils.Spectrum1D`.
A specific example of this is demonstrated here.

Note that the workflow described here is for spectral cubes that are rectified
such that one of the axes is entirely spectral and all the spaxels have the same
``spectral_axis`` values (i.e., case 2 in :ref:`specutils-representation-overview`).
For less-rectified cubes, pre-processing steps (not addressed by specutils at the

> time of this writing) will be necessary to rectify the cubes into that form.

Reviewer 2: ```suggestion
time of this writing) will be necessary to rectify the cubes into that form.
Note, though, that such cubes can be stored in specutils data structures (cases
3 and 4 in :ref:`specutils-representation-overview`), which support *some* of 
these behaviors, even though the fulkl set of tools do not yet apply.
```
